.github/workflows/minimum.yml

```
@@ -24,6 +24,19 @@ jobs:
       uses: actions/setup-python@v5
       with:
         python-version: ${{ matrix.python-version }}
+    - name: Cache pip on windows
+      id: cache-pip
+      if: runner.os == 'Windows'
+      uses: actions/cache@v3
+      with:
+        path: C:\Users\runneradmin\AppData\Local\pip\Cache
+        key: ${{ runner.os }}-pip-${{ matrix.python-version }}-${{ hashFiles('**/pyproject.toml') }}
+        restore-keys: |
+            ${{ runner.os }}-pip-${{ matrix.python-version }}-
+    - name: Check cache hit
+      if: runner.os == 'Windows'
+      run: |
+        echo "Cache hit status: ${{ steps.cache-pip.outputs.cache-hit }}"
     - name: Install dependencies
       run: |
```

You can use the cache to determine how to install dependencies:
```suggestion
    - name: Install SDV (cache has not changed)
      if: steps.cache-pip.outputs.cache-hit == 'true'
      run:  |
        python -m pip install --no-dependencies .
    - name: Install SDV and dependencies (cache has changed)
      if: steps.cache-pip.outputs.cache-hit != 'true'
      run: |
```

---

Cannot call the force command `--no-dependencies`, as that command ignores installing dependencies altogether ([Job example](https://github.com/sdv-dev/SDV/actions/runs/9878701812/job/27283222021)). 

Pip looks into the cache location path and will directly install those files rather than re-download them so we don't need to have separate steps. Example [here](https://github.com/sdv-dev/SDV/actions/runs/9876951679/job/27277353093): we can see it using the cache and cut down the time for windows installation by not downloading. (~2-3 mins saved)